lib/services/nativescript-migration-service.ts

```diff
@@ -85,92 +117,185 @@ export class NativeScriptMigrationService implements IFrameworkMigrationService
+			}
+		}).future<void>()();
+}
+
 	private downloadNativeScriptMigrationFile(): IFuture<void> {
 		let remoteFilePath = `${this.remoteNativeScriptResourcesPath}/${NativeScriptMigrationService.REMOTE_NATIVESCRIPT_MIGRATION_DATA_FILENAME}`;
 		return this.$resourceDownloader.downloadResourceFromServer(remoteFilePath, this.$nativeScriptResources.nativeScriptMigrationFile);
 	}
 
-	private downloadTnsModules(language: string, version: string): IFuture<void> {
-		let fileName = this.getFileNameByVersion(version);
-		let remotePathUrl = `${this.remoteNativeScriptResourcesPath}/tns_modules/${language}/${fileName}`;
-		let filePath = path.join(this.tnsModulesDirectoryPath, language, fileName);
-		return this.$resourceDownloader.downloadResourceFromServer(remotePathUrl, filePath);
+	private downloadTnsPackage(language: string, version: string): IFuture<void> {
+		if (language === NativeScriptMigrationService.TYPESCRIPT_ABBREVIATION || semver.lt(version, this.nativeScriptMigrationData.wait().modulesNpmMinimumVersion)) {
+			let fileName = this.getFileNameByVersion(version);
+			let remotePathUrl = `${this.remoteNativeScriptResourcesPath}/${NativeScriptMigrationService.TNS_MODULES}/${language}/${fileName}`;
+			let filePath = path.join(this.tnsModulesDirectoryPath, language, fileName);
+			return this.$resourceDownloader.downloadResourceFromServer(remotePathUrl, filePath);
+		}
+
+		return Future.fromResult();
 	}
 
 	private getFileNameByVersion(version: string): string {
-		return `${version}.zip`;
+		let fileSuffix = semver.gte(version, this.nativeScriptMigrationData.wait().modulesNpmMinimumVersion) ? NativeScriptMigrationService.TNS_CORE_MODULES : "";
+		return `${version}${fileSuffix}.zip`;
 	}
 
-	private getObsoleteVersions(): IFuture<string[]> {
+	private getDeletedVersions(): IFuture<string[]> {
 		return ((): string[] => {
 			let migrationData = this.nativeScriptMigrationData.wait();
-			return _.map(migrationData.obsoleteVersions, obsoleteVersion => obsoleteVersion.version);
+			return _.map(migrationData.deletedVersions, deletedVersion => deletedVersion.version);
 		}).future<string[]>()();
 	}
 
-	/**
-	 * Checks if the provided {N} version is obsolete and fails with correct error message in such case.
-	 * If the version is not marked as obsolete, do nothing.
-	 * @param {string} version The version tha has to be checked.
-	 * @returns {IFuture<void>}
-	 */
-	private checkIsVersionObsolete(version: string): IFuture<void> {
+	private warnIfVersionObsolete(version: string): IFuture<void> {
+		return (() => {
+			if (!_.contains(this.getSupportedVersions().wait(), version) && !this.isDeleted(version).wait()) {
+				this.$logger.warn("Your project's version is deprecated and will become obsolete in a future release. You can still develop and build your project but it is recommended that you commit all changes to version control and migrate to a newer version of NativeScript.");
```

Comment:
"Your project version" instead of "Your project's version". Comma before but is not needed - the punctuation rules in English are not always identical to the rules in Bulgarian.

And why is it that I'm supplying a message once and then seeing someone's improvisation on the same message across multiple PRs? As far as I can remember initially this was  "Your project targets NativeScript [version]. This version is deprecated and..."